crown-court-contribution/src/main/java/uk/gov/justice/laa/crime/contribution/service/MaatCalculateContributionService.java

```diff
@@ -57,17 +57,13 @@ private static boolean isUpliftApplied(CalculateContributionDTO calculateContrib
                 calculateContributionDTO.getDateUpliftRemoved() == null;
     }
 
-    public static BigDecimal getAnnualDisposableIncome(final CalculateContributionDTO calculateContributionDTO, final BigDecimal annualDisposableIncome) {
-        if (annualDisposableIncome == null) {
-            if ((calculateContributionDTO.getDisposableIncomeAfterMagHardship() != null)) {
-                return calculateContributionDTO.getDisposableIncomeAfterMagHardship();
-            } else {
-                if (calculateContributionDTO.getTotalAnnualDisposableIncome() != null) {
-                    return calculateContributionDTO.getTotalAnnualDisposableIncome();
-                } else return BigDecimal.ZERO;
-            }
+    public static BigDecimal getAnnualDisposableIncome(final CalculateContributionDTO calculateContributionDTO) {
```

Comment:
I thunk it is better to use if-else as against nested ternary operator as it makes code easier to read and maintain and makes it explicit that the value is defaulted to Zero

BigDecimal disposableIncome;
if (calculateContributionDTO.getDisposableIncomeAfterMagHardship() != null) {
    disposableIncome = calculateContributionDTO.getDisposableIncomeAfterMagHardship();
} else if (calculateContributionDTO.getTotalAnnualDisposableIncome() != null) {
    disposableIncome = calculateContributionDTO.getTotalAnnualDisposableIncome();
} else {
    disposableIncome = BigDecimal.ZERO;
}
return disposableIncome;


Just a different choice, so approving it....

Reply:
Yeh fair point, amended the statement to the above to make it easier to understand.